tests/test_logging.py

```diff
@@ -195,6 +196,62 @@ def some_route():
         in mock_req_logger.log.call_args_list
     )
 
+    assert (
+        mock.call(
+            mock.ANY,
+            AnyStringMatching(r".*streaming response.*", flags=re.I),
+            mock.ANY,
+            extra=mock.ANY,
+        )
+        not in mock_req_logger.log.call_args_list
+    )
+
+    time.sleep(0.05)
```

Comment:
I'm not sure I understand what this `assert` and `sleep` are for

Reply:
The "streaming response closed" log message should only be emitted after the response is consumed/closed, so the assert first checks that it hasn't already been emitted before we've done that.

The sleep is because this test also covers the behaviour of the `request_time` field: the `after_request` log message should have a value of at least 0.05s and then the `call_on_close` log message should be at least 0.05s after that, i.e. at least 0.1s.

Reply:
Cool, makes sense, thanks!